Automattic/jetpack
# Sync: Move 'active_modules' to default whitelisted callables

As `active_modules` callable is heavily used by standalone plugins and no longer depends on the Jetpack plugin, it makes sense to move them to the list of default whitelisted Sync callables so we ensure it's always synced without forcing the consumers to explicitly add it to the list.

#### Changes proposed in this Pull Request:
* `Jetpack`: Remove `active_modules` from plugin's added callables
* `Sync\Functions`: Introduce `get_active_modules` method
* `Sync\Defaults`: Add `active_modules` to Sync's default whitelisted callables

#### Jetpack product discussion
pdWQjU-o-p2
p1652965539642009-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Jetpack plugin: Try enabling/disabling modules and confirm the change is reflected on the Cache site ( `get_option( 'jetpack_active_modules' )` )
- Jetpack Social: Activate the plugin and make sure `publicize` is in the list of active modules on the Cache site
- Jetpack Search: Activate the plugin and make sure `search` is in the list of active modules on the Cache site
